src/router/responder.rs

```diff
@@ -0,0 +1,172 @@
+use super::{response, ResponseBody, ResponseType};
+pub use hyper::{Body, Error, HeaderMap, Request, Response, StatusCode};
+
+use serde::ser::Serialize;
+
+pub type ResponseResult<T = Response<Body>> = Result<T, http::Error>;
+
+pub trait Responder {
+    fn respond_to(self) -> ResponseResult;
+    fn with_status(self, status: StatusCode) -> CustomResponder<Self>
+    where
+        Self: ResponseBody + Sized,
+    {
+        CustomResponder::new(self).with_status(status)
+    }
+}
+
+/// Allows to override status code and headers for a responder.
+pub struct CustomResponder<T> {
+    body: T,
+    status: Option<StatusCode>,
+    // headers: Option<HeaderMap>,
+    // error: Option<Error>,
+}
+
+impl<T> CustomResponder<T>
+where
+    T: ResponseBody,
+{
+    fn new(body: T) -> Self {
+        CustomResponder {
+            body,
+            status: None,
+            // headers: None,
+            // error: None,
+        }
+    }
+
+    pub fn with_status(mut self, status: StatusCode) -> Self {
+        self.status = Some(status);
+        self
+    }
+}
+
+impl<T> Responder for CustomResponder<T>
+where
+    T: ResponseBody,
+{
+    fn respond_to(self) -> ResponseResult {
+        let status = match self.status {
+            Some(status) => status,
+            None => StatusCode::OK,
+        };
+        Response::builder()
+            .status(status)
+            .body(self.body.into_body())
+    }
+}
+
+impl Responder for String {
```

Comment:
Able to use macro for all primitive type.

Reply:
ya, will do it